### PR TITLE
feat(zipkin) override unknown-service-name

### PR DESCRIPTION
**Pull request by jeremyjpj0916**
Optional configurable field to enable setting a defaulted service name in the zipkin spans to replace the unknown-service-name existing behavior,

Fixes: https://github.com/Kong/kong-plugin-zipkin/issues/39
Fixes: https://github.com/Kong/kong-plugin-zipkin/issues/23  (I think if I am reading it correctly)

I defaulted the string to nil as to not change any behavior of the plugin by default. Could easily set default to "Kong" if that is preferred but that would change default behavior so I didn't commit it like that.

### CODE REVIEW

File: kong/plugins/zipkin/schema.lua [hunk 1]
```diff
@@ -11,6 +11,8 @@ return {
 					{ sample_ratio = { type = "number",
 					                   default = 0.001,
 					                   between = { 0, 1 } } },
+					{ peer_service_base = { type = "string",
```
Comment (james-callahan) [inline kong/plugins/zipkin/schema.lua]: What do you mean by "base"?

Comment (jeremyjpj0916) [reply kong/plugins/zipkin/schema.lua]: Maybe base is not the best word, but its what simply gets populated in the unknown-service-name field in the screenshots of one of the issues: https://user-images.githubusercontent.com/124244/52150794-51f9dc00-2671-11e9-8e6e-00949df2794f.png .
I am not attached to the name by any means. could be called just ```service_name``` or whatever is fancied.

**Comment by devinsba** [comment]
Sorry for the drive-by but as a zipkin contributor I watch a lot of repos to see how people are using it. I would disagree with this change. I don't think a proxy/lb/api-gateway should set `localServiceName` to the name of the service it is proxying to, I would expect `localServiceName` to be something like `internal-apps-api-gateway` or something to that affect.

Ultimately what I'm saying is I don't think this is the correct way to fix #39 within the zipkin model, and I agree with the original poster of #23 on the way forward being a constant for the entire instance

cc @adriancole

**Comment by james-callahan** [comment]
> Sorry for the drive-by but as a zipkin contributor I watch a lot of repos to see how people are using it. I would disagree with this change. I don't think a proxy/lb/api-gateway should set `localServiceName` to the name of the service it is proxying to, I would expect `localServiceName` to be something like `internal-apps-api-gateway` or something to that affect.

That's not what this PR does.
Rather, it allows the user to configure, on a per zipkin plugin basis, what the service should be called.
I imagine the common configuration would be to configure the plugin globally, which gives the effect you mention.

**Comment by jeremyjpj0916** [comment]
Yeah I specifically avoided setting it to the service name of where Kong is proxying too. We get that in the access phase logged already. This is a static config as @james-callahan described above.

**Comment by devinsba** [comment]
Ok, then I was confused by the `peer` portion of the variable name looks like what I would expect then

**Comment by jeremyjpj0916** [comment]
Yeah, I am open to alternative variable names, wasn't sure whats best. Maybe ```default_service_name``` or something.

**Comment by jeremyjpj0916** [comment]
Oof, I done goofed in my commit squashing now I think 😆

**Comment by jeremyjpj0916** [comment]
Closing this one, continuing over here: https://github.com/Kong/kong-plugin-zipkin/pull/44

**Pull request closed without merging**